pkg/virt-controller/services/template.go

```diff
@@ -760,10 +764,35 @@ func (t *templateService) newResourceRenderer(vmi *v1.VirtualMachineInstance, ne
 		return nil, err
 	}
 
-	options := append(baseOptions, t.VMIResourcePredicates(vmi, networkToResourceMap).Apply()...)
+	memoryLimitsRatio := t.getMemoryLimitsRatio(vmi)
+	options := append(baseOptions, t.VMIResourcePredicates(vmi, networkToResourceMap, memoryLimitsRatio).Apply()...)
 	return NewResourceRenderer(vmiResources.Limits, vmiResources.Requests, options...), nil
 }
 
+func (t *templateService) getMemoryLimitsRatio(vmi *v1.VirtualMachineInstance) float64 {
```

Comment:
BTW, my main reasoning for this is not necessarily the performance penalty, which is suspected to be relatively low because of the informer.  It is rather that we should isolate Alpha features as much as possible and avoid executing 
alpha-related code when the feature gate is disabled.

Reply:
This makes totally sense! thank you!